src/datasets/iterable_dataset.py

```diff
@@ -1350,8 +1351,10 @@ def _concatenate_iterable_datasets(
     else:
         _check_column_names([col_name for dset in dsets for col_name in dset.features])
 
+    # TODO: improve this to account for a mix of ClassLabel and Value for example
+    # right now it would keep the type of the first dataset in the list
     features = Features()
-    for dset in dsets:
+    for dset in dsets[::-1]:
```

Comment:
We should use `_align_features` (defined in `arrow_dataset.py`) here to keep this logic aligned with the arrow-based dataset.